plugins/linux/ifplugin/linuxcalls/dump_interface_linuxcalls.go

```diff
@@ -0,0 +1,353 @@
+					ParentIndex:   link.Attrs().ParentIndex,
+					MasterIndex:   link.Attrs().MasterIndex,
+					OperState:     uint8(link.Attrs().OperState),
+					Flags:         link.Attrs().RawFlags,
+					Encapsulation: link.Attrs().EncapType,
+					NumRxQueues:   link.Attrs().NumRxQueues,
+					NumTxQueues:   link.Attrs().NumTxQueues,
+					TxQueueLen:    link.Attrs().TxQLen,
+				},
+			})
+
+			// build interface statistics
+			retrieved.stats = append(retrieved.stats, &InterfaceStatistics{
+				Name:         iface.Name,
+				Type:         iface.Type,
+				LinuxIfIndex: link.Attrs().Index,
+				RxPackets:    link.Attrs().Statistics.RxPackets,
+				TxPackets:    link.Attrs().Statistics.TxPackets,
+				RxBytes:      link.Attrs().Statistics.RxBytes,
+				TxBytes:      link.Attrs().Statistics.TxBytes,
+				RxErrors:     link.Attrs().Statistics.RxErrors,
+				TxErrors:     link.Attrs().Statistics.TxErrors,
+				RxDropped:    link.Attrs().Statistics.TxDropped,
+				TxDropped:    link.Attrs().Statistics.RxDropped,
+			})
+		}
+
+		// switch back to the default namespace
+		revert()
+	}
+
+	ch <- retrieved
+}
+
+// retrieveLinkDetails retrieves link details common to all interface types (e.g. addresses).
+func (h *NetLinkHandler) retrieveLinkDetails(link netlink.Link, iface *interfaces.Interface, nsRef *namespaces.NetNamespace) {
+	var err error
+	// read interface status
+	iface.Enabled, err = h.IsInterfaceUp(link.Attrs().Name)
+	if err != nil {
+		h.log.WithFields(logging.Fields{
+			"if-host-name": link.Attrs().Name,
+			"namespace":    nsRef,
+		}).Warn("Failed to read interface status:", err)
+	}
+
+	// read assigned IP addresses
+	addressList, err := h.GetAddressList(link.Attrs().Name)
+	if err != nil {
+		h.log.WithFields(logging.Fields{
+			"if-host-name": link.Attrs().Name,
+			"namespace":    nsRef,
+		}).Warn("Failed to read address list:", err)
+	}
+	for _, address := range addressList {
+		if address.Scope == unix.RT_SCOPE_LINK {
+			// ignore link-local IPv6 addresses
+			continue
+		}
+		mask, _ := address.Mask.Size()
+		addrStr := address.IP.String() + "/" + strconv.Itoa(mask)
+		iface.IpAddresses = append(iface.IpAddresses, addrStr)
+	}
+
+	// read checksum offloading
+	if iface.Type == interfaces.Interface_VETH {
+		rxOn, txOn, err := h.GetChecksumOffloading(link.Attrs().Name)
+		if err != nil {
+			h.log.WithFields(logging.Fields{
+				"if-host-name": link.Attrs().Name,
+				"namespace":    nsRef,
+			}).Warn("Failed to read checksum offloading:", err)
+		} else {
+			if !rxOn {
+				iface.GetVeth().RxChecksumOffloading = interfaces.VethLink_CHKSM_OFFLOAD_DISABLED
+			}
+			if !txOn {
+				iface.GetVeth().TxChecksumOffloading = interfaces.VethLink_CHKSM_OFFLOAD_DISABLED
+			}
+		}
+	}
+}
+
+// parseVethAlias parses out VETH logical name together with the peer name from the alias.
+func parseVethAlias(alias string) (vethName, peerName string) {
+	aliasParts := strings.Split(alias, "/")
+	vethName = aliasParts[0]
+	if len(aliasParts) > 0 {
```

Comment:
looks like it should be `if len(aliasParts) > 1 {`

Reply:
Good catch